packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -1133,23 +1139,26 @@ export class ProjectLifecycleManager {
       return metaState
     }
 
-    const configFileTs = this._pathToFile('cypress.config.ts')
-    const configFileJs = this._pathToFile('cypress.config.js')
+    for (const fileName of potentialConfigFiles) {
+      const filePath = this._pathToFile(fileName)
+      const fileExists = fs.existsSync(filePath)
 
-    if (fs.existsSync(configFileTs)) {
-      metaState.hasValidConfigFile = true
-      this.setConfigFilePath('cypress.config.ts')
-    }
+      if (fileExists) {
+        // We'll collect all the found config files.
+        // If we found more than one, this list will be used in an error message.
+        metaState.allFoundConfigFiles.push(fileName)
 
-    if (fs.existsSync(configFileJs)) {
-      metaState.hasValidConfigFile = true
-      if (this._configFilePath) {
-        metaState.hasMultipleConfigPaths = true
-      } else {
-        this.setConfigFilePath('cypress.config.js')
+        // We've found our first config file! We'll continue looping to make sure there's
+        // only one. Looping over all config files is done so we can provide rich errors and warnings.
+        if (!this._configFilePath) {
+          metaState.hasValidConfigFile = true
```

Comment:
This seems weird - if there is NOT configFilePath, hasValidConfigFile is true? Is this condition backwards?

Reply:
If we haven't set the configFilePath yet, this is the first one. This is the same conditional and logic that existed before my change.

Reply:
Right... assume valid if we haven't found one

This is really confusing and we should revisit this soon

Reply:
🤷🏻 it's a common looping pattern imho

```js
let found = false

for (item of items) {
  // do things to each item

  // do things to first item found
  if (!found && someConditional(item)) {
    found = true
    doStuff(item)
  }
}
```